app/controllers/users_controller.rb

```diff
@@ -118,7 +118,8 @@ def reservations
   # GET /facilities/:facility_id/users/:user_id/accounts
   def accounts
     # accounts for this facility
-    @account_users = @user.account_users.active
+    facility_accounts = Account.for_facility(current_facility)
+    @account_users = @user.account_users.active.select { |ua| facility_accounts.include?(ua.account) }
```

Comment:
Ok, I Changed #for_facility to be a scope on Account and moved the specs to the model. Also, I added a spec to check that deleted accounts are filtered.